pallets/delegation/src/delegation_hierarchy.rs

```diff
@@ -89,12 +102,17 @@ impl<T: Config> DelegationNode<T> {
 		hierarchy_root_id: DelegationNodeIdOf<T>,
 		parent: DelegationNodeIdOf<T>,
 		details: DelegationDetails<T>,
+		deposit_owner: AccountIdOf<T>,
 	) -> Self {
 		Self {
 			hierarchy_root_id,
 			parent: Some(parent),
 			children: BoundedBTreeSet::<DelegationNodeIdOf<T>, T::MaxChildren>::new(),
 			details,
+			deposit: Deposit::<AccountIdOf<T>, BalanceOf<T>> {
+				owner: deposit_owner,
+				amount: <T as Config>::Deposit::get(),
```

Comment:
M(reserve-deposit part 3): if we hard code the deposit here, we never made sure that the deposit was reserved somewhere. So we might end up to store a deposit without reserving.

Reply:
Not sure what your solution proposal would look like after moving the reserving into `store_delegation_under_parent` and `create_and_store_new_hierarchy`.

Reply:
Please have a look at f384e3.

Reply:
The commented out version looks like the thing I meant. 👍🏻

Reply:
Great! Hope this issue is resolved by fdbf92d then?

Reply:
Love it!